fdbrpc/FlowTransport.h

```diff
@@ -44,7 +44,9 @@ class Endpoint {
 	}
 
 	void choosePrimaryAddress() {
-		if(addresses.secondaryAddress.present() && !g_network->getLocalAddresses().secondaryAddress.present() && (addresses.address.isTLS() != g_network->getLocalAddresses().address.isTLS())) {
+		if(addresses.secondaryAddress.present() && 
+		((!g_network->getLocalAddresses().secondaryAddress.present() && (addresses.address.isTLS() != g_network->getLocalAddresses().address.isTLS())) ||
+		(g_network->getLocalAddresses().secondaryAddress.present() && !addresses.address.isTLS()))) {
```

Comment:
Is this new change saying that if there are two addresses in g_network->getLocalAddresses, then that means we should be using TLS, and so swap if we aren't?

Reply:
yes, this is fixing a bug where two copies of the same endpoint would not compare as equal because the network addresses were in a different order